File: GenerateFirstNFibonacciNumbers.py
#!/usr/bin/env python3
def main():
    for x in range(1000):
        print("Fibonacci %4d: %50d" % (x, fib(x)))
def fib(x, d = {}):
    if x in d:
        return d[x]
    else:
        if x == 0:
            d[x] = 0
            return 0
        elif x == 1:
            d[x] = 1
            return 1
        else:
            v = fib(x-1) + fib(x-2)
            d[x] = v
            return v

if __name__ == "__main__": main()

File: Generate First N Fibonacci Numbers.c
#include <stdio.h>
#define N 45
int main(){
    long int Fib[N], i;
    Fib[0] = 1;
    Fib[1] = 1;
    for(i = 2; i<N; i++){
        Fib[i] = Fib[i-1] + Fib[i-2];
    }
    for(i=0; i<N; i++){
        printf("%ld ", Fib[i]);
    }
}
